lib/clock.ex

```diff
@@ -1,128 +1,58 @@
 defmodule UUIDv7.Clock do
   @moduledoc false
-  use GenServer
 
-  @type timestamp :: integer()
-  @type counter :: <<_::18>>
-  @type counter_seed :: <<_::17>>
-
-  @default_table_opts [:named_table, :public, :set]
-  @default_cleanup_interval_ms :timer.seconds(2)
-  @default_cleanup_tick_cutoff 2
-
-  # The threshold number where the counter will roll over.
-  @max_counter 2 ** 18
-
-  @compile {:inline, [update_counter: 2]}
-
-  @doc """
-  Get a unix millisecond timestamp and a clock sequence that fits in 18 bits.
-  """
-  @spec next(counter_seed()) :: {timestamp(), counter()}
-  def next(<<seed::17>>) do
-    timestamp_ref = :persistent_term.get(__MODULE__)
-    previous_ts = :atomics.get(timestamp_ref, 1)
-    current_ts = System.system_time(:millisecond)
-
-    # Time leap backwards protection.
-    current_ts = if current_ts < previous_ts, do: previous_ts, else: current_ts
-
-    :atomics.put(timestamp_ref, 1, current_ts)
+  # For macOS (Darwin) or Windows use 10 bits.
+  # Otherwise, use 12 bits (e.g. on Linux).
+  step_bits =
+    case :os.type() do
+      {:unix, :darwin} -> 10
+      {:win32, _} -> 10
+      {_, _} -> 12
+    end
```

Comment:
That would be problematic when cross-compiling.

Reply:
people successfully cross-compile?

Reply:
i guess I could check at runtime :(